src/vs/workbench/contrib/notebook/browser/contrib/statusBar/contributedStatusBarItemController.ts

```diff
@@ -35,12 +35,20 @@ export class ContributedStatusBarItemController extends Disposable implements IN
 	}
 
 	private _updateEverything(): void {
-		this._visibleCells.forEach(cell => cell.dispose());
-		this._visibleCells.clear();
-		this._updateVisibleCells({ added: this._observer.visibleCells, removed: [] });
+		const newCells = this._observer.visibleCells.filter(cell => !this._visibleCells.has(cell.handle));
+		const visibleCellHandles = new Set(this._observer.visibleCells.map(item => item.handle));
+		const currentCellHandles = Array.from(this._visibleCells.keys());
+		const removedCells = currentCellHandles.filter(handle => !visibleCellHandles.has(handle));
+		const itemsToUpdate = currentCellHandles.filter(handle => visibleCellHandles.has(handle));
+
+		this._updateVisibleCells({ added: newCells, removed: removedCells.map(handle => ({ handle })) });
```

Comment:
E.g. if a cell was deleted, we need to delete that corresponding item from `this._visibleCells`, then only way we know if its been deleted or not longer visible is if it doesn't exist in `this._observer.visibleCells`, at that point, we're trying to remove an item from `this.visibleCells`, and we don't have the CellViewModel to pass that into `_updateVisibleCells`. I could update the types to store the CelLViewModel as well, or the like, but that feels unnecessary.